test/reparent.py

```diff
@@ -621,6 +621,60 @@ def test_reparent_with_down_slave(self, shard_id='0'):
     tablet.kill_tablets([tablet_62344, tablet_62044, tablet_41983,
                          tablet_31981])
 
+  def test_change_type_semi_sync(self):
+    utils.run_vtctl(['CreateKeyspace', 'test_keyspace'])
+
+    # create the database so vttablets start, as they are serving
+    tablet_62344.create_db('vt_test_keyspace')
+    tablet_62044.create_db('vt_test_keyspace')
+    tablet_41983.create_db('vt_test_keyspace')
+    tablet_31981.create_db('vt_test_keyspace')
+
+    # Start up a soon-to-be master, 3 replicas mysql and vttablet
+    tablet_62344.init_tablet('replica', 'test_keyspace', '0', start=True,
+                             wait_for_start=False)
+    tablet_62044.init_tablet('replica', 'test_keyspace', '0', start=True,
+                             wait_for_start=False)
+    tablet_41983.init_tablet('replica', 'test_keyspace', '0', start=True,
+                             wait_for_start=False)
+    tablet_31981.init_tablet('rdonly', 'test_keyspace', '0', start=True,
+                             wait_for_start=False)
+    for t in [tablet_62344, tablet_62044, tablet_41983, tablet_31981]:
+      t.wait_for_vttablet_state('NOT_SERVING')
+
+    # Force the slaves to reparent assuming that all the datasets are
+    # identical.
+    utils.run_vtctl(['InitShardMaster', '-force', 'test_keyspace/0',
+                     tablet_62344.tablet_alias], auto_log=True)
+    utils.validate_topology(ping_tablets=True)
+    self._check_master_tablet(tablet_62344)
+
+    # Stop replication on one slave, to make sure when we make it
+    # rdonly it doesn't start.
+    utils.run_vtctl(['StopSlave', tablet_41983.tablet_alias])
```

Comment:
This part doesn't seem to be related since you don't use tablet_41983 after this and don't make it rdonly.